tests/status/ignore.c

```diff
@@ -1213,3 +1213,41 @@ void test_status_ignore__unignored_subdirs(void)
 	assert_is_ignored("dir/a.test");
 	refute_is_ignored("dir/subdir/a.test");
 }
+
+void test_status_ignore__ignored_subdirfiles_with_subdir_rule(void)
+{
+	static const char *test_files[] = {
+    "empty_standard_repo/dir/a.test",
+		"empty_standard_repo/dir/sub1/sub2/b.test",
+		NULL
+	};
```

Comment:
There's no need to create those testfiles in "test/attr/ignore.c". You can just rewrite the gitignore file and then immediately use `assert_is_ignored` in there.